### DISCUSSION TITLE
Use FastLogin on my plugin

### DISCUSSION THREAD

**Discussion by murillofrazaocunha** [games647/FastLogin#1226] [General]
Hello, good evening. I’m trying to create a plugin for Velocity, and I’d like to use your library to check if a player is premium. If they are, I want to redirect them to one server; if not, I want to redirect them to another. However, I’m not sure how to implement this check using FastLogin in the AuthenticateEvent. The authentication plugin is custom, and I don’t want it to require a command to activate premium status. What I want is to simply verify whether the player is premium or not automatically using your API and apply it in the AuthenticateEvent of Velocity

**Comment by TuxCoding** [comment]
>The authentication plugin is custom, and I don’t want it to require a command to activate premium status

That's hard, because the detection is based on the username at the moment. If we request online mode authentications from offline players, the client will just disconnect. [Reference](https://github.com/games647/FastLogin/wiki/FAQ#why-do-players-have-to-invoke-a-command)

**Reply by murillofrazaocunha** [reply]
Ah, right, I had forgotten about that. So how could I integrate FastLogin with my plugin, even if it's just with commands?

**Reply by TuxCoding** [reply]
First you need to provide the auth plugin hook implementation for FastLogin [ref](https://github.com/games647/FastLogin/blob/4dd6b9ade434f5765a7a473c80e99d7f1d929a84/core/src/main/java/com/github/games647/fastlogin/core/shared/FastLoginCore.java#L303). Then it will fire `AutoLoginEvent`s and will invoke the `forceLogin` method once the connection was verified.
